### PR TITLE
doctr

### PR DESCRIPTION
**Pull request by ocefpaf**
This is the first PR to create the auto-publishing feature.

@emiliom gave me `adim` rights to this repo, so I already added the [ssh key](https://github.com/BiG-CZ/notebook_data_demo/settings/keys) and activated `Travis-CI`.

@lsetiawan let's coordinate Monday on a strategy for `AppVeyor`.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,7 +10,12 @@ cache:
 # at $HOME/miniconda/pkgs. That is a win-win because when re-creating the
 # env we will download only the new packages.
 before_cache:
-  - rm -rf $HOME/miniconda/envs/IOOS
+  - rm -rf $HOME/miniconda/envs/BiG-CZ
+  - rm -rf $HOME/miniconda/envs/PAGE
```
Comment (ocefpaf) [inline .travis.yml:14]: We want to cache the packages not the envs, that way we save time when the packages are the same by downloading only what is new.

File: .travis.yml [hunk 2]
```diff
@@ -10,7 +10,12 @@ cache:
...
+
+env:
+  global:
+    - secure: "FdmfFPlylPvVz5f+YIIpk1YnwndJoM8s+yyg16hxlycorfTpOaFA+em1TlUFqUWIAa0dc7JWgc0L90ui/Usv41un1Cli6nbG+3rM/Mb/vCSMQCVcbsyB9eFBJNSXmokwB7qtvVzVSXBCOGajLTNLefr7bT+AYzRwd7TaZWKgXJ04D7aRhcKp7nyRkc2ooWdWlpzsK7X9YRvTbWD2ghFHxB70JuA1Zv7VzluxFUJfrKTBwsjVzcZMYXekCjxJiXcIsnOHCZ7ybGlB9v63RvXhOrXcx4Iq7ONGCoeTGKk/8+eMOXYOoXP7tV4WGiltBODqT7zt15YNRSRAjLW7zTXpCJoDi154/F7pbafzJcanV7kOgAtjXMDBS0Krvy9OFokC2zcF2DeOrqWk1rDpnL2KqopsPjSlWDnzpd+bYL9dB5IvVeWvBHWYssjySnj+drrXFbYHx0uxgY62Ok9KiIY5X4QbgGA8ND7bEk2/FFmgAX9I44TxjXLk9P5Y1ka/yaR0eH/8FqdLJqkesMbCkiVz5UnDCAZgarN9bkOMpHHOLHNlS9E10wttiWXK1NbYQxGLgwj5zkuFbzPesYfdZxXqCcFSbZT+iNUJNLQJc/iYQ+bLpSEkQZytMoqmHlhFZRbT82Iu3pXChu7FA6MML5uL28YoYoxv6If2sl9vi+FZsOM="
```
Comment (ocefpaf) [inline .travis.yml:18]: @lsetiawan this is the encrypted ssh key.

File: .travis.yml [hunk 3]
```diff
@@ -46,26 +53,33 @@ before_install:
     conda update conda
 
 install:
-  - conda env create --file environment.yml --name BiG-CZ
-  - source activate BiG-CZ
+  # We need to envs b/c the BiG-CZ software stack is not compatible with Pyhton 3 yet.
+  - conda env create --file webpage/environment.yml --name PAGE
+  - conda env create --file environment.yml --name BiG-CZ ;
```
Comment (ocefpaf) [inline .travis.yml:58]: We need two envs b/c the publishing software is Python 3 only while most of the BiG-CZ stack is still Python 2.7 only.

File: .travis.yml [hunk 4]
```diff
@@ -46,26 +53,33 @@ before_install:
...
 
   # Test python notebooks.
   - if [ $TEST_TARGET == "default" ]; then
+      source activate BiG-CZ ;
       python tests/test_notebooks.py ;
     fi
 
   # Publish the notebooks.
   - if [ $TEST_TARGET == "publish" ]; then
+      source activate BiG-CZ ;
```
Comment (ocefpaf) [inline .travis.yml:79]: I could activate only once but I decided to be explicit when we are using each env.

**Comment by emiliom** [comment]
This is exactly how I would've done it :wine_glass: ...

**Pull request merged**